packages/react-button/src/components/MenuButton/MenuButton.tsx

```diff
@@ -1,15 +1,30 @@
-import { compose, createClassResolver } from '@fluentui/react-compose';
+import * as React from 'react';
 import { ChevronDownIcon } from '@fluentui/react-icons';
-import { ContextualMenu } from 'office-ui-fabric-react';
-import { MenuButtonBase } from './MenuButtonBase';
+import { ContextualMenu, useFocusRects } from 'office-ui-fabric-react';
+import { createMenuButton } from './createMenuButton';
 import { MenuButtonProps } from './MenuButton.types';
 import * as classes from './MenuButton.scss';
+import { useButtonClasses } from '../Button/Button';
+import { makeClasses } from '@fluentui/react-compose/lib/next/index';
+import { useInlineTokens } from '@fluentui/react-theme-provider';
 
-export const MenuButton = compose<'button', MenuButtonProps, MenuButtonProps, {}, {}>(MenuButtonBase, {
-  classes: createClassResolver(classes),
-  displayName: 'MenuButton',
-  slots: {
-    menu: ContextualMenu,
-    menuIcon: ChevronDownIcon,
-  },
+export const useMenuButtonClasses = makeClasses(classes);
+
+export const MenuButton = React.forwardRef<HTMLElement, MenuButtonProps>((props, ref) => {
+  const { state, render } = createMenuButton(props, ref, {
+    menuIcon: { as: ChevronDownIcon },
+    menu: { as: ContextualMenu },
+  });
+
+  // Styling hooks.
+  useButtonClasses(state);
+  useMenuButtonClasses(state);
+  useFocusRects(state.ref);
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  useInlineTokens(state as any);
```

Comment:
why is `as any` needed here?

Reply:
I've added a comment on the error but i don't get why. will see if I can resolve with you offline.

Reply:
State in `useInlineTokens` is expected to be:

```jsx
{ style?: React.CSSProperties; tokens?: TokenSetType }
```
The object we're passing in of type `MenuButtonState` should indeed have these.

Typescript errors with:
> Type 'MenuButtonState' has no properties in common with type '{ style?: CSSProperties | undefined; tokens?: TokenSetType | undefined; }'.

Now, the state type should indeed have these well defined:

```
export type MenuButtonProps = Omit<ButtonProps, 'iconPosition' | 'loader'> & {
```

Where `ButtonProps` is defined as :

```jsx
export type ButtonProps = ComponentProps &
  React.HTMLAttributes<HTMLElement> & {
    // ...
    tokens?: RecursivePartial<ButtonTokenSet>;
};
```
TypeScript things that neither style or tokens are valid in this type. I'm stumped here.